srml/executive/src/lib.rs

```diff
@@ -355,6 +358,19 @@ where
 					return TransactionValidity::Invalid(ApplyError::CantPay as i8)
 				}
 
+				// pay and burn the tip if provided.
+				let tip = xt.tip();
+				let tip_value = match tip {
```

Comment:
Might be worth moving to `Tip` implementation So that you can simplify this to `let tip_vaue = xt.tip().value()`?